docker-compose.yml

```diff
@@ -732,6 +732,8 @@ services:
     shm_size: *shm-size
     environment:
       <<: *ccache
+      ARROW_R_CXXFLAGS: '-Werror'
+      LIBARROW_BUILD: 'false'
```

Comment:
What does this do?

Reply:
This disables the part of the R package configure script that would build Arrow C++ libraries if they weren't found. In the `ubuntu-r` job, we build the C++ library outside of the R package installation, so we already have it and should be testing against that build. This is just a failsafe in case for some reason we break the ability to use that C++ library built outside the R package installation, to make sure the build fails hard.